src/components/worker/WorkerConfigForm.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useReducer, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import Select from 'react-select';
+
+import { Worker, WorkerOption } from './workerInterfaces';
+import { reducer, initialState } from './workerReducer';
+import workersApi from '../../api/workersAPI';
+
+const WorkerConfigForm = () => {
+  const history = useHistory();
+  const [state, dispatch] = useReducer(
+    reducer,
+    initialState,
+    () => initialState
+  );
+
+  const {
+    firstName,
+    lastName,
+    preferredShift,
+    preferredCoworkers,
+    workers,
+    options,
+  } = state;
+
+  const createOptions = (workersList: Worker[]) => {
+    const opt: WorkerOption[] = [];
+    workersList.forEach((worker) => {
+      opt.push({
+        value: worker.id,
+        label: worker.public_id
+          ? `${worker.first_name} ${worker.last_name} (${worker.public_id})`
+          : `${worker.first_name} ${worker.last_name}`,
+      });
+    });
+
+    return opt;
+  };
+
+  const addWorker = () => {
+    workersApi
+      .addWorker({
+        first_name: firstName,
+        last_name: lastName,
+        preferred_shift: preferredShift,
+        preferred_coworkers: preferredCoworkers,
+      })
+      .then((res) => res)
+      .catch((err) => {
+        throw err;
+      });
+  };
+
+  const handleChange = (
```

Comment:
As far as I can see, you can move this function to "global hook tools", because this is kind of universal in the context of useReducer with forms. BTW this is a bummer. In the VUE it was much better with those compouted properties.

Reply:
Yup. Vue 1, React 0.